xds/src/main/java/io/grpc/xds/internal/rbac/engine/GrpcAuthorizationEngine.java

```diff
@@ -361,12 +363,11 @@ private String deserializeHeader(String headerName) {
         if (values == null) {
           return null;
         }
-        List<String> decoded = new ArrayList<>();
+        List<String> encoded = new ArrayList<>();
         for (byte[] v : values) {
-          String encoded = new String(v, US_ASCII);
-          decoded.add(new String(BaseEncoding.base64().decode(encoded), US_ASCII));
+          encoded.add(BaseEncoding.base64().encode(v));
```

Comment:
We need `omitPadding()` as well, per https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md